types/vinyl-fs/index.d.ts

```diff
@@ -160,60 +118,38 @@ export function src(globs: string|string[], opt?: SrcOptions): NodeJS.ReadWriteS
  * cwd, base, and path will be overwritten to match the folder
  * stat.mode will be overwritten if you used a mode parameter
  * contents will have it's position reset to the beginning if it is a stream
- * @param folder destination folder
- */
-export function dest(folder: string, opt?: DestOptions): NodeJS.ReadWriteStream;
-
-/**
- * On write the stream will save the vinyl File to disk at the folder/cwd specified.
- * After writing the file to disk, it will be emitted from the stream so you can keep piping these around.
- * The file will be modified after being written to this stream:
- * cwd, base, and path will be overwritten to match the folder
- * stat.mode will be overwritten if you used a mode parameter
- * contents will have it's position reset to the beginning if it is a stream
- * @param getFolderPath function that takes in a file and returns a folder path
+ * @param folder destination folder or a function that takes in a file and returns a folder path
  */
-export function dest(getFolderPath: (file: File) => string): NodeJS.ReadWriteStream;
+export function dest(folder: string | ((file: File) => string), opt?: DestOptions): NodeJS.ReadWriteStream;
```

Comment:
The signature passing a function to get the folder path also supports options, which was missing previously. And this means it becomes a union type in the `folder` argument rather than an overload.